src/main/java/com/conveyal/gtfs/loader/JdbcGtfsLoader.java

```diff
@@ -380,9 +386,16 @@ private int loadInternal (Table table) throws Exception {
             // The first field holds the line number of the CSV file. Prepared statement parameters are one-based.
             if (postgresText) transformedStrings[0] = Integer.toString(lineNumber);
             else insertStatement.setInt(1, lineNumber);
+
+            // Remove the values from the csv row for any bad fields (i.e., duplicate headers or named "ID").
+            // CSV values needs to be a LinkedList to support removal operation (Arrays.asList is read-only).
+            List<String> values = new LinkedList<>(Arrays.asList(csvReader.getValues()));
```

Comment:
@abyrd, I considered doing this, but got tripped up because I failed to consider that I could just maintain a separate column index for setting the prepared statement parameters (it must account for those null fields).  There's an easy way to do that though.  Instead of using `f` as the index, I'll just maintain a separate one.

Reply:
I still need some version of the `rawFields` array though because I need to create some other array with the clean set of fields to construct the Table definition.

Reply:
I didn't think of this before. So you need both something to tell you which fields to skip and something to tell you which ones are present. Another idea: add an `int csvColumn` field to the `Field` class, make and iterate over a list of only the fields you want to load (indexes within that list are the prepared statement parameters), and draw the column number from the new field. But I'd lean toward iterating over the list of Fields with nulls in it, filtering out the nulls at the moment it's used in the Table definition.